lib/babe/babe.go

```diff
@@ -68,7 +68,101 @@ type ServiceConfig struct {
 	Telemetry          telemetry.Client
 }
 
-// NewService returns a new Babe Service using the provided VRF keys and runtime
+// Validate returns error if config does not contain required attributes
+func (sc *ServiceConfig) Validate() error {
+	if sc.Keypair == nil && sc.Authority {
+		return errNoBABEAuthorityKeyProvided
+	}
+
+	if sc.BlockState == nil {
+		return ErrNilBlockState
+	}
+
+	if sc.EpochState == nil {
+		return errNilEpochState
+	}
+
+	if sc.BlockImportHandler == nil {
+		return errNilBlockImportHandler
+	}
+
+	return nil
+}
+
+// ServiceIFace interface that defines methods available to BabeService
+type ServiceIFace interface {
+	Start() error
+	Stop() error
+	EpochLength() uint64
+	Pause() error
+	IsPaused() bool
+	Resume() error
+	SlotDuration() uint64
+}
+
+// ServiceBuilder interface to define the building of babe service
+type ServiceBuilder interface {
+	NewServiceIFace(cfg *ServiceConfig) (ServiceIFace, error)
+}
+
+var _ ServiceBuilder = (*Builder)(nil)
```

Comment:
The ServiceBuilder interface was created so that we have an interface for generating mocks for that, mocks for this interface are used in service_test.go Test_nodeBuilder_createBABEService tests.  I'm not sure how to generate mocks without an interface.

Reply:
You could have used an anonymous function signature instead of an interface. But you would have to manually write a dummy function that implemented that function signature for testing. 

I don't think this interface needs to be defined in the `babe` package.  You could move this interface to be defined in `dot` to localise it to `nodeBuilder.createBABEServiceWithBuilder`.  It doesn't really make sense for the `babe` package to have to have an interface for one if it's types constructors. Can we change this?

Reply:
@edwardmack Have you addressed this?